strats.md

```diff
@@ -45,3 +47,411 @@ Execution of a strat may have an effect of clearing one or more [obstacles](regi
+
+#### Example
+```json
+{
+  "name": "Leave With Spark",
+  "notable": false,
+  "requires": [
+    {"canShinecharge": {
+      "usedTiles": 20,
+      "openEnd": 0
+    }},
+    {"shinespark": {
+      "frames": 200
+    }}
+  ],
+  "exitCondition": {
+    "leaveWithSpark": {}
+  }
+}
+```
+
+## Entrance conditions
+
+In all strats with an `entranceCondition`, the `from` node of the strat must be a door node. An `entranceCondition` object must contain exactly one property:
+
+- _comeInRunning_: This indicates that Samus must run into the room, with speed in a certain range.
+- _comeInJumping_: This indicates that Samus must run and jump just before hitting the transition, with speed in a certain range.
+- _comeInCharging_: This indicates that Samus must run into the room with enough space to complete a shinecharge.
+- _comeInCharged_: This indicates that Samus must enter the room with a shinecharge with a certain amount of frames remaining.
+- _comeInWithSpark_: This indicates that Samus must shinespark into the room.
+- _comeInWithBombBoost_: This indicates that Samus must come into the room with a horizontal bomb boost.
+- _comeInStutterCharging_: This indicates that Samus must run into the room with a stutter immediately before the transition.
+- _comeInWithDoorStuckSetup_: This indicates that Samus must enter the room in a way that allows getting stuck in the door as it closes.
+
+Each of these properties is described in more detail below.
+
+### Come In Running
+
+An `comeInRunning` entrance condition represents the need for Samus to be able to run into the room with speed in a certain range. It has the following properties:
+* _speedBooster_: If true, then Speed Booster must be used while running into the room. If false, then Speed Booster must not be used. If "any", then Speed Booster may or may not be used.
+* _minTiles_: The minimum horizontal speed that will satisfy the condition, measured in effective runway tiles with dash held.
+* _maxTiles_: The maximum horizontal speed that will satisfy the condition, measured in effective runway tiles with dash held.
+
+A `comeInRunning` condition can be satisfied only by a matching strat on the other side of the door with `leaveWithRunway` exit condition: a match is valid if all of the following are true:
+- The effective runway length of the `leaveWithRunway` is at least as long as the `minTiles` in the `comeInRunning` condition.
+
+Where applicable, a `comeInRunning` condition also includes implicit requirements for actions to be performed in the previous room, which are effectively prepended to the start of the strat's `requires` (or equivalently but more properly, onto the end of the `requires` of the `leaveWithRunway` strat in the other room):
+- If `speedBooster` is true, then there is an implicit "SpeedBooster" item requirement.
+- If the previous room is heated, then `heatFrames` are required based on the time needed to perform the run. 
```

Comment:
Might be worth clarifying - the heat frames are implicit, and gravity below is not?

Reply:
These are all implicit requirements. Clarified the Gravity one though now, to make it clear that it depends on the entrance condition in the next room (since Gravity isn't always needed)